google-cloud-logging/src/main/java/com/google/cloud/logging/MonitoredResourceUtil.java

```diff
@@ -144,135 +206,152 @@ public static MonitoredResource getResource(String projectId, String resourceTyp
    * @return custom log entry enhancers
    */
   public static List<LoggingEnhancer> getResourceEnhancers() {
-    Resource resourceType = getAutoDetectedResourceType();
+    Resource resourceType = detectResourceType();
     return createEnhancers(resourceType);
   }
 
+  @SuppressWarnings("incomplete-switch")
   private static String getValue(Label label, String resourceType) {
-    String value;
+    String value = "";
+
```

Comment:
This whole method will need to be refactored (doesn't need to be in this PR). The switch...case will become really long really soon (more detected environments from 220+ list). I'll review it assuming it stays this way for now but please open a task on this repo to refactor it to be more sane.

Reply:
It can be refactored using visitor pattern, where each node will represent a managed resource with its labels. Example here: https://en.wikipedia.org/wiki/Visitor_pattern#Java_example This way we won't have any "ifs" or "switch" statements and instead have a bunch of classes that represent resources and a method per class that "visits" (detects) the resource. IMO, it's a lot cleaner and easier to test, esp. if we want to expose this logic to other use-cases, such as metrics collection, OpenTelemetry Trace exporters, etc. 

There might be other alternatives available. It's not something we need to do in this PR but I want us to do this "refactoring cleanup" once, so adding more resources will be a lot easier going forward.